Kubernetes actions

frank-spano opened the pull request:
### What does this PR do?

The cluster-agent subscribes to the K8S_ACTIONS Remote Configuration product on startup when kubeactions.enabled is true. When a config arrives, the ConfigRetriever acknowledges receipt via RC's applyStateCallback and passes it to the ActionProcessor. The processor validates the action: checking the timestamp is within the 1-minute TTL, confirming the action type is set, and running action-specific validation (e.g. delete_pod requires namespace and a resource_id for UID verification). An in-memory ActionStore tracks processesd actions to prevent duplicate execution.

  Validated actions are dispatched through an ExecutorRegistry to the appropriate executor, which uses the client-go typed Kubernetes client to call the API server. After execution, the result is acknowledged back to RC and sent to EVP.

### Motivation

### Describe how you validated your changes

### Additional Notes
testing rbac
```
# Additional RBAC for kubeactions - allows pod deletion and deployment restarts
# This creates a separate ClusterRole with the necessary permissions for testing
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: datadog-cluster-agent-kubeactions
  labels:
    app: datadog-cluster-agent
rules:
  # Allow deleting pods
  - apiGroups: [""]
    resources: ["pods"]
    verbs: ["delete", "get", "list"]
  # Allow restarting deployments (via patch to trigger rollout)
  - apiGroups: ["apps"]
    resources: ["deployments"]
    verbs: ["get", "list", "patch"]
  # Allow reading deployment status
  - apiGroups: ["apps"]
    resources: ["deployments/status"]
    verbs: ["get"]
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: datadog-cluster-agent-kubeactions
  labels:
    app: datadog-cluster-agent
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: datadog-cluster-agent-kubeactions
subjects:
  - kind: ServiceAccount
    name: dd-datadog-cluster-agent
    namespace: default
```

bitomaxsp commented on comp/forwarder/eventplatform/eventplatformimpl/epforwarder.go:
```diff
@@ -293,6 +293,19 @@ func getPassthroughPipelines() []passthroughPipelineDesc {
 			defaultBatchMaxSize:           pkgconfigsetup.DefaultBatchMaxSize,
 			defaultInputChanSize:          500,
 		},
+		// TODO: Add kubeactions EVP pipeline once the intake endpoint is provisioned
```
Is this for future PR, or it will be added in this PR ?

wdhif reviewed (changes requested):
Hello 👋 
Can you add `pkg/clusteragent/kubeactions/` to the `.github/CODEOWNERS` and mark your team as owners?
Thanks!

bitomaxsp commented on pkg/clusteragent/kubeactions/config_retriever.go:
```diff
@@ -0,0 +1,100 @@
+	}
+}
+
+func min(a, b int) int {
```
IIRC `min` is builtin in go

frank-spano committed:
Add Kubernetes Actions feature for cluster-agent

The cluster-agent subscribes to the K8S_ACTIONS Remote Configuration product
when kubeactions.enabled is true. Validated actions are dispatched to typed
executors (delete_pod, restart_deployment, patch_deployment) which use the
client-go Kubernetes API. Results are acknowledged back to RC.

Gated behind kubeactions.enabled (default: false). No impact on clusters
where the feature is not explicitly enabled.

JSGette reviewed (approved):
OK for scope of the build team

wdhif reviewed (approved):
Approved for @DataDog/container-platform files

dustmop reviewed (approved):
LGTM for agent-configuration owned files

frank-spano committed:
update test
